src/esp/bindings_js/bindings_js.cpp

```diff
@@ -51,6 +51,20 @@ std::map<std::string, ObservationSpace> Simulator_getAgentObservationSpaces(
   return spaces;
 }
 
+template <class T, typename... Targs>
```

Comment:
@mosra You are probably the best person to review this part. Do we really not have any better bindings exposed via Magnum?

Reply:
¯\\\_(ツ)_/¯

Magnum has no JS bindings on its own. Everything has been happening organically here so far, on an as-needed basis.

Reply:
FYI, I've reached out to Blaise about making some refactors here. The JS code uses Eigen types (vec4f, etc.), perhaps for legacy reasons.

Reply:
@morsa does Magnum::Quaternion really not have a constructor for Vec4f though?

Reply:
Ah nvm, @eundersander posted the answer below.

Reply:
There's a conversion [from/to Eigen's Quaternion](https://doc.magnum.graphics/magnum/GeometryIntegration_8h.html) class, and possibly Eigen itself could have some way to convert to a 4-component vector on its own.

I'm not providing a conversion to a 4-component vector because it's not clear whether it should be xyzw or wxyz -- each math library uses something else, so it's better to be explicit. Which means doing a conversion using `data()[i]` also might not be 100% safe -- I don't know what order is the `vec4f` expected to be in and if it matches the order inside Magnum's Quaternion.